packages/server/graphql/mutations/helpers/notifications/MSTeamsNotifier.ts

```diff
@@ -0,0 +1,423 @@
+    summaryColumn.addItem(summaryTextBlock)
+    summaryColumnSet.addColumn(summaryColumn)
+    card.addItem(summaryColumnSet)
+
+    const meetingEndedActionsColumnSet = new AdaptiveCards.ColumnSet()
+    meetingEndedActionsColumnSet.spacing = AdaptiveCards.Spacing.ExtraLarge
+    const meetingEndedDiscussionColumn = new AdaptiveCards.Column()
+    meetingEndedDiscussionColumn.width = 'auto'
+    const meetingEndedReviewColumn = new AdaptiveCards.Column()
+    meetingEndedReviewColumn.width = 'auto'
+
+    const meetingEndedDiscussionActionSet = new AdaptiveCards.ActionSet()
+    const meetingEndedDiscussionAction = new AdaptiveCards.OpenUrlAction()
+    meetingEndedDiscussionAction.title = 'See discussion'
+    meetingEndedDiscussionAction.url = meetingUrl
+    meetingEndedDiscussionAction.id = 'joinMeeting'
+
+    meetingEndedDiscussionActionSet.addAction(meetingEndedDiscussionAction)
+    meetingEndedDiscussionColumn.addItem(meetingEndedDiscussionActionSet)
+
+    const meetingEndedReviewActionSet = new AdaptiveCards.ActionSet()
+    const meetingEndedReviewAction = new AdaptiveCards.OpenUrlAction()
+    meetingEndedReviewAction.title = 'Review summary'
+    meetingEndedReviewAction.url = summaryUrl
+    meetingEndedReviewAction.id = 'joinMeeting'
+
+    meetingEndedDiscussionActionSet.addAction(meetingEndedReviewAction)
+    meetingEndedDiscussionColumn.addItem(meetingEndedReviewActionSet)
+
+    meetingEndedActionsColumnSet.addColumn(meetingEndedDiscussionColumn)
+    meetingEndedActionsColumnSet.addColumn(meetingEndedReviewColumn)
+
+    card.addItem(meetingEndedActionsColumnSet)
+
+    const adaptivecard = JSON.stringify(card.toJSON())
+    const attachments = `{"type":"message", "attachments":[{"contentType":"application/vnd.microsoft.card.adaptive","contentUrl":null, "content": ${adaptivecard}}]}`
+
+    return notifyMSTeams('meetingEnd', webhookUrl, facilitatorUserId, team.id, attachments)
+  },
+
+  async startTimeLimit(scheduledEndTime, meeting, team) {
+    const {webhookUrl} = notificationChannel
+
+    const {phases, facilitatorStageId, facilitatorUserId} = meeting
+
+    const maybeMeetingShortLink = makeAppURL(process.env.INVITATION_SHORTLINK!, `${meeting.id}`)
+    const meetingUrl = makeAppURL(appOrigin, `meet/${meeting.id}`)
+
+    const stageRes = findStageById(phases, facilitatorStageId)
+    const {stage} = stageRes!
+    const {phaseType} = stage
+    const phaseLabel = phaseLabelLookup[phaseType]
+
+    const meetingTitle = `The **${phaseLabel} Phase** has begun ⌛`
+
+    const card = new AdaptiveCards.AdaptiveCard()
+    card.version = new AdaptiveCards.Version(1.2, 0)
+
+    const titleTextBlock = GenerateACMeetingTitle(meetingTitle)
+    card.addItem(titleTextBlock)
+
+    const meetingDetailColumnSet = GenerateACMeetingAndTeamsDetails(team, meeting)
+    card.addItem(meetingDetailColumnSet)
+
+    let teamfixedtime = scheduledEndTime.toISOString()
+    teamfixedtime = teamfixedtime.replace(/.\d+Z$/g, 'Z')
```

Comment:
+1 rather than using a `let`, we could do `const teamFixedTime = scheduledEndTime.toISOString().replace(/.\d+Z$/g, 'Z')`